javascript/index.js

```diff
@@ -4,37 +4,152 @@
-  }, (error) => console.log(error));
+// getInstruction("mashedPotatoes", 0, (step1) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+// }, (error) => console.log(error));
+
+// getInstruction("mashedPotatoes", 1, (step2) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+// }, (error) => console.log(error));
+
+// getInstruction("mashedPotatoes", 2, (step3) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+// }, (error) => console.log(error));
 
+// getInstruction("mashedPotatoes", 3, (step4) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+// }, (error) => console.log(error));
 
+// getInstruction("mashedPotatoes", 4, (step5) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+//   document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+// }, (error) => console.log(error));
 
 // Iteration 1 - using callbacks
-// ...
+getInstruction("mashedPotatoes", 0, (step0) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`;
+  getInstruction("mashedPotatoes", 1, (step1) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+    getInstruction("mashedPotatoes", 2, (step2) => {
+      document.querySelector(
+        "#mashedPotatoes"
+      ).innerHTML += `<li>${step2}</li>`;
+      getInstruction("mashedPotatoes", 3, (step3) => {
+        document.querySelector(
+          "#mashedPotatoes"
+        ).innerHTML += `<li>${step3}</li>`;
+        getInstruction("mashedPotatoes", 4, (step4) => {
+          document.querySelector(
+            "#mashedPotatoes"
+          ).innerHTML += `<li>${step4}</li>`;
+          document
+            .querySelector("#mashedPotatoesImg")
+            .removeAttribute("hidden");
+          document.querySelector(
+            "#mashedPotatoes"
+          ).innerHTML += `<li>Mashed potatoes are ready!</li>`;
+        });
+      });
+    });
+  });
+});
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0).then((step0) => {
+  document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+  obtainInstruction("steak", 1).then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    obtainInstruction("steak", 2).then((step2) => {
+      document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+      obtainInstruction("steak", 3).then((step3) => {
+        document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+        obtainInstruction("steak", 4).then((step4) => {
+          document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+          obtainInstruction("steak", 5).then((step5) => {
+            document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+            obtainInstruction("steak", 6).then((step6) => {
```

Comment:
Here you are using the `.then()` but you are calling it inside the function like the previous iteration. The correct way of doing this is be returning the `obtainInstruction `function:
```
obtainInstruction("steak", 0)
  .then((step0) => {
    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
    return obtainInstruction("steak", 1);
  })
  .then((step1) => {
    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
    return obtainInstruction("steak", 2);
  })

( ... )
```